setup.py

```diff
@@ -203,6 +204,56 @@ def run(self):
             sys.exit(128)
 
 
+class Selftests(SimpleCommand):
+    """Run selftests"""
+
+    description = 'Run selftests'
+    user_options = [
+        ("job-api", None, "Run job API checks"),
+        ("static-checks", None, "Run static checks (isort, lint, etc)"),
+        ("selftests-nrunner-interface", None, "Run selftests/functional/test_nrunner_interface.py"),
+        ("selftests-unit", None, "Run selftests/unit/"),
+        ("selftests-jobs", None, "Run selftests/jobs/"),
+        ("selftests-functional", None, "Run selftests/functional/"),
+        ("selftests-optional-plugins", None, "Run optional_plugins/*/tests/"),
+        ("disable-plugin-checks", None, "Disable checks for a plugin (by directory name)"),
+        ("features", None, "Show the features tested by this test"),
+        ("all", None, "all")
+    ]
+
+    def initialize_options(self):
+        self.job_api = False  # pylint: disable=W0201
+        self.static_checks = False  # pylint: disable=W0201
+        self.selftests_nrunner_interface = False  # pylint: disable=W0201
+        self.selftests_unit = False  # pylint: disable=W0201
+        self.selftests_jobs = False  # pylint: disable=W0201
+        self.selftests_functional = False  # pylint: disable=W0201
+        self.selftests_optional_plugins = False  # pylint: disable=W0201
+        self.disable_plugin_checks = []  # pylint: disable=W0201
+        self.features = False  # pylint: disable=W0201
+        self.all = False  # pylint: disable=W0201
```

Comment:
I'm wondering if this shouldn't be `True` by default. Any reason for that?

Reply:
> I'm wondering if this shouldn't be `True` by default. Any reason for that?

Yes, because by default this option isn't selected by the user unless they type `--all`. To be honest, the the option is cosmetic right now, both `python3 setup.py selftests` and `python3 setup.py selftests --all` will run all the tests. But I think eventually we might default to something running only some tests if nothing is provided instead of running all of them. Thoughts?

Reply:
If `python3 setup.py selftests` will be the same as `python3 setup.py selftests --all` than the default for `--all` should be true, right? And maybe we don't need `--all` at all.

Would be nice to avoid having multiple commands to run all our tests.